src/lexer.coffee

```diff
@@ -314,55 +314,90 @@ exports.Lexer = class Lexer
   # everything has been parsed and the JavaScript code generated.
   commentToken: (chunk = @chunk) ->
     return 0 unless match = chunk.match COMMENT
-    [comment, here] = match
+    [withLeadingWhitespace, hereLeadingWhitespace, here, hereTrailingWhitespace, nonHere] = match
```

Comment:
There's a fair amount of cleanup/preservation of comment location data going on here

Reply:
Sure, updated to use `hereComment` and `lineComment` instead (which are much better names, you're correct about what they contain)